torch_spyre/csrc/spyre_mem.cpp

```diff
@@ -44,6 +48,411 @@
 
 namespace spyre {
 
+struct DMAParameters {
+  const int64_t size_bytes;  // bytes to transfer
+  const off64_t src_offset;
+  const off64_t dst_offset;
+};
+auto get_device_shape(c10::IntArrayRef sizes) -> std::vector<int64_t> {
+  /* Given the CPU Tensor's shape, return the shape of the tensor on Spyre
+   */
```

Comment:
this assumes the tensor on cpu is contiguous right? this is why we ignore the strides completely. If so, please make it clear on the comment

Reply:
Yes. @ani300 : maybe we should give a test case for non-contiguous tensor, and marked as skipped so that we can test that in the future. Do you have one suggestion of such tensor?

Reply:
@ani300 : we added a TODO to support strides tensors.